module/src/hooks/useIntersectionObserver.ts

```diff
@@ -3,28 +3,59 @@ import * as React from 'react';
 import { Globals } from '../utils/globals';
 
 /**
- * Use an intersection observer to fire the passed callback upon intersection - also cleans up on unmount
+ * Use an intersection observer to fire the passed callback - also cleans up on unmount. Can either be used by just passing in a ref, or by using the functions returned to observe and disconnect
  * @param ref the html element to watch
  * @param callback the callback to be fired
- * @param options an intersection observer options object
+ * @param options options for the intersection observer
  */
 
 export function useIntersectionObserver(
-  ref: React.MutableRefObject<HTMLElement>,
-  callback: (isIntersecting: boolean, entries: IntersectionObserverEntry[], io: IntersectionObserver) => any,
-  options?: IntersectionObserverInit
+  callback: IntersectionObserverCallback,
+  options?: IntersectionObserverInit,
+  ref?: React.MutableRefObject<Element | undefined | null>
 ) {
   const observer = React.useRef<IntersectionObserver>();
 
+  const observe = React.useCallback(
+    (element: Element) => {
+      observer.current = new IntersectionObserver(callback, options);
+      observer.current.observe(element);
+    },
+    [callback, options]
+  );
+
+  const unobserve = React.useCallback((element: Element) => {
+    if (observer.current) {
+      observer.current.unobserve(element);
+    }
+  }, []);
+
+  const disconnect = React.useCallback(() => {
+    observer.current?.disconnect();
+  }, []);
+
   React.useLayoutEffect(() => {
     if (!!ref && !!ref.current && Globals.isBrowser && Globals.supportsIntersectionObserver) {
```

Comment:
Unnecessary `!!ref`, refs are always objects.

Reply:
The ref is optional on the hook, as it can be used with the callbacks returned which you can handle on your own.